library-integration-slack/src/test/kotlin/library/integration/slack/services/slack/SlackMessageAccessorIntegrationTest.kt

```diff
@@ -0,0 +1,117 @@
+package library.integration.slack.services.slack
+
+import com.github.tomakehurst.wiremock.client.WireMock
+import com.github.tomakehurst.wiremock.client.WireMock.*
+import library.integration.slack.services.error.handling.ErrorHandler
+import org.assertj.core.api.Assertions.*
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.MethodSource
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.context.TestConfiguration
+import org.springframework.cloud.contract.wiremock.AutoConfigureWireMock
+import org.springframework.context.annotation.ComponentScan
+import org.springframework.http.HttpStatus.*
+import org.springframework.test.context.ContextConfiguration
+import org.testit.testutils.logrecorder.api.LogRecord
+import org.testit.testutils.logrecorder.junit5.RecordLoggers
+import utils.classification.IntegrationTest
+import utils.extensions.EnableSpringExtension
+import utils.services.error.handling.ErrorHandlerDataProvider.Companion.slackChannelArchivedException
+import utils.services.error.handling.ErrorHandlerDataProvider.Companion.slackChannelNotFoundException
+import utils.services.error.handling.ErrorHandlerDataProvider.Companion.slackChannelProhibitedException
+import utils.services.error.handling.ErrorHandlerDataProvider.Companion.slackInvalidPayloadException
+import utils.services.error.handling.ErrorHandlerDataProvider.Companion.slackServerException
+import java.util.stream.Stream
+
+
+@SpringBootTest
+@IntegrationTest
+@EnableSpringExtension
+@AutoConfigureWireMock(port = 9999)
+@ContextConfiguration(classes = [TestConfiguration::class])
+class SlackMessageAccessorIntegrationTest {
+
+    @ComponentScan(basePackageClasses = [SlackMessageAccessor::class, ErrorHandler::class])
+    class TestConfiguration
+
+    @Autowired
+    lateinit var slackSettings: SlackSettings
+
+    @Autowired
+    lateinit var slackMessageAccessor: SlackMessageAccessor
+
+    @Autowired
+    lateinit var slackMessageClient: SlackMessageClient
+
+    @Autowired
+    lateinit var errorHandler: ErrorHandler
+
+    val slackMessage = "some msg"
+
+    @BeforeEach
+    fun setUp() {
+        // FIXME: Add dynamic port handling
+        slackSettings.baseUrl = "http://localhost:9999"
```

Comment:
You can get the dynamic port by setting `@AutoConfigureWireMock(port = 0)` and autowiring an instance of `WireMockServer` and using it's `getPort()` method here. (`slackSettings.baseUrl = "http://localhost:${wireMock.getPort()}"`)

Reply:
Thanks for the tip, I guess that was the missing piece 👍 It is updated now.